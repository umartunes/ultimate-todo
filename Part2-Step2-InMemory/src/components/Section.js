import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { deleteTodo, editStatus } from '../redux/actions/actions-todos'

class Section extends React.Component {

    loopItems = () => {

        const { todos, dispatch } = this.props

        return todos.map((todo, i) => {

            return <React.Fragment key={i}>
                <div className="row card taskListBorder" >
                    <div className="col s1">
                        <label>
                            <input type="checkbox" defaultChecked={todo.status !== 'pending'} className="filled-in" onChange={() => dispatch(editStatus(todo._id))} />
                            <span></span>
                        </label>
                    </div>
                    <div className="col s8 truncate">
                        {todo.status === 'pending' ? todo.title : (<s>{todo.title}</s>)}

                    </div>
                    <div className="col s3 right-align red-text tex-darken-4 right-align">
                        <button className="btn-small red waves-effect waves-lighten hoverable" onClick={() => dispatch(deleteTodo(todo._id))} ><FontAwesomeIcon icon="trash-alt" size="1x" /> Delete</button>
                    </div>
                </div>
            </React.Fragment>

        })

    }

    render() {

        return (
            <React.Fragment>

                <main style={{ margin: 10 }} >
                    {this.props.todos ? this.loopItems() : <p>There are no tasks in your list</p>}
                </main>

                {/* <!-- Fixed Floating Action Button --> */}
                <div className="fixed-action-btn">
                    <Link to="/create" className="btn-floating btn-large waves-effect waves-light hoverable sky-blue">
                        <FontAwesomeIcon icon="plus" size="lg" />
                    </Link>
                </div>
            </React.Fragment>
        )
    }
}

const mapStateToProps = (state) => {

    return {
        todos: state.todos.todos
    }
}

export default connect(mapStateToProps)(Section)
