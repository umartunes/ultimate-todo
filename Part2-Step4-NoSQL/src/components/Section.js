import React from 'react'
import { Link } from 'react-router-dom'
import TaskList from './TaskList';
import { connect } from 'react-redux'
import { fetch, updateSatus } from '../redux/actions/actions-todos';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
class Section extends React.Component {

    componentWillMount() {

        this.props.fetch()
    }

    render() {
        return (
            <React.Fragment>

                <main className="tod-List-Setting" >
                    {this.props.todos && this.props.todos.length > 0 ? this.props.todos.map((todo, i) => {

                        return <TaskList key={i} todo={todo} status={todo.status} updateSatus={this.props.updateSatus} />

                    }) : <p>No Todo List</p>}

                </main>

                {/* <!-- Fixed Floating Action Button --> */}
                <div className="fixed-action-btn">
                    <Link to="/create" className="btn-floating btn-large waves-effect waves-light hoverable sky-blue">
                        <FontAwesomeIcon icon="plus" size="1x" />
                    </Link>
                </div>
            </React.Fragment>
        )
    }
}
const getTodoList = (state) => {

    return {
        todos: state.todos.todos
    }
}

export default connect(getTodoList, { fetch, updateSatus })(Section)
