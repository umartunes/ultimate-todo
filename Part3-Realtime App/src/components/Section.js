import React from 'react'
import { Link } from 'react-router-dom'
import TodoList from './TodoList';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

class Section extends React.Component {

    render() {
        return (
            <React.Fragment>
                <main className="tod-List-Setting" >
                    <TodoList />
                </main>
                <div className="fixed-action-btn">
                    <Link to="/create" className="btn-floating btn-large waves-effect waves-light hoverable sky-blue">
                        <FontAwesomeIcon icon="plus" size="1x" />
                    </Link>
                </div>
            </React.Fragment>
        )
    }
}
export default Section
