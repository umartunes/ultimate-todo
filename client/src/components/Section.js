import React from 'react'
import { Link } from 'react-router-dom'
import TaskList from './TaskList';
import { connect } from 'react-redux'
import { fetch } from '../redux/actions/actions-todos';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
class Section extends React.Component {

    componentWillMount() {

        this.props.fetch()
    }
    render() {
        return (
            <React.Fragment>

                <main style={{ margin: 10 }} >
                    {this.props.todos && this.props.todos.length > 0 ? this.props.todos.map((todo, i) => {
                        return <TaskList key={i} todo={todo} />

                    }) : <p>No TaskLis</p>}


                </main>

                {/* <!-- Fixed Floating Action Button --> */}
                <div className="fixed-action-btn">
                    <Link to="/create" className="btn-floating btn-large waves-effect waves-light hoverable sky-blue">
                     <FontAwesomeIcon icon="plus" size="1x" />
                    </Link>
                </div>
            </React.Fragment>
        )
    }
}
const getTodoList = (state) => {

    return {
        todos: state.todos.todos
    }
}

export default connect(getTodoList, { fetch })(Section)
