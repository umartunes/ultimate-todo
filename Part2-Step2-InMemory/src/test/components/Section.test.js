import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Section from '../../components/Section';




test('Should Render Section Page Correctly',()=>{
    
    const renderer = new ReactShallowRenderer();
    renderer.render(<Section/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
})