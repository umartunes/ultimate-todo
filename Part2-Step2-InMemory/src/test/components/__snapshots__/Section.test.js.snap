// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Section Page Correctly 1`] = `
<Unknown>
  <main
    className="tod-List-Setting"
  >
    <Connect(TodoList) />
  </main>
  <div
    className="fixed-action-btn"
  >
    <Link
      className="btn-floating btn-large waves-effect waves-light hoverable sky-blue"
      replace={false}
      to="/create"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="1x"
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Link>
  </div>
</Unknown>
`;
